### PR TITLE
[GOBBLIN-2051] Rework `FlowLaunchHandler`, `DagActionStore`, and related classes for clarity

### PR DESCRIPTION
**Pull request by phet**
Dear Gobblin maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [ ] My PR addresses the following [Gobblin JIRA](https://issues.apache.org/jira/browse/GOBBLIN/) issues and references them in the PR title. For example, "[GOBBLIN-XXX] My Gobblin PR"
    - https://issues.apache.org/jira/browse/GOBBLIN-2051

### Description
- [ ] Here are some details about my PR, including screenshots (if applicable):

Rework portions of `FlowLaunchHandler`, `DagActionStore`, and related classes for clarity, including:
- add: `DagActionStore::addDagAction(DagAction)`
- name explicitly: `LeaseAttemptStatus::getConsensusDagAction()`
- migrate to `java.util.Optional`

### Tests
- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

updated existing unit tests

### Commits
- [ ] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

File: gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManagementTaskStreamImpl.java [hunk 1]
```diff
@@ -177,7 +177,7 @@ private DagTask createDagTask(DagActionStore.DagAction dagAction, LeaseAttemptSt
   */
   protected void scheduleReminderForEvent(LeaseAttemptStatus leaseStatus)
       throws SchedulerException {
-    dagActionReminderScheduler.get().scheduleReminder(leaseStatus.getDagAction(),
+    dagActionReminderScheduler.get().scheduleReminder(leaseStatus.getLaunderedDagAction(),
```
Comment (umustafi) [inline gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManagementTaskStreamImpl.java:180]: This renaming is helpful to make the laundering more explicit

File: gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/FlowLaunchHandler.java [hunk 2]
```diff
@@ -99,51 +104,47 @@ public FlowLaunchHandler(Config config,
...
-        // If persisting the dag action failed, then we set another trigger for this event to occur immediately to
-        // re-attempt handling the event
-        scheduleReminderForEvent(jobProps,
-            new LeaseAttemptStatus.LeasedToAnotherStatus(leaseObtainedStatus.getDagAction(), 0L), eventTimeMillis);
-      } else if (leaseAttemptStatus instanceof LeaseAttemptStatus.LeasedToAnotherStatus) {
-        scheduleReminderForEvent(jobProps, (LeaseAttemptStatus.LeasedToAnotherStatus) leaseAttemptStatus,
-            eventTimeMillis);
-      }
-      // Otherwise leaseAttemptStatus instanceof MultiActiveLeaseArbiter.NoLongerLeasingStatus & no need to do anything
+    // TOOD - (needs correcting?) semantically, isn't - `skipFlowExecutionIdReplacement == !adoptConsensusFlowExecutionId`?!?!?
```
[the remark this answers is not in the document]
Comment (umustafi) [reply gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/FlowLaunchHandler.java:110]: let's keep it consistent so we don't accidentally put the wrong value -> we can keep `adoptConsensusFlowExecutionId` across the board.

**Review by umustafi — APPROVED** [review]
great changes overall! let's change the boolean used for laundering (or not) to be consistent semantically across usage

**Review by umustafi — APPROVED** [review]

File: gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/FlowLaunchHandler.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (phet) [reply gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/FlowLaunchHandler.java]: sg.  for consistency, i also renamed `launderedDagAction` to `consensusDagAction`

**Review by Will-Lo — APPROVED** [review]

**Pull request merged**